pkg/cmd/repo/view/view.go

```diff
@@ -0,0 +1,211 @@
+package view
+
+import (
+	"errors"
+	"fmt"
+	"html/template"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/utils"
+	"github.com/spf13/cobra"
+)
+
+type ViewOptions struct {
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	RepoArg string
+	Web     bool
+}
+
+func NewCmdView(f *cmdutil.Factory, runF func(*ViewOptions) error) *cobra.Command {
+	opts := ViewOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		BaseRepo:   f.BaseRepo,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "view [<repository>]",
+		Short: "View a repository",
+		Long: `Display the description and the README of a GitHub repository.
+
+With no argument, the repository for the current directory is displayed.
+
+With '--web', open the repository in a web browser instead.`,
+		Args: cobra.MaximumNArgs(1),
+		RunE: func(c *cobra.Command, args []string) error {
+			if len(args) > 0 {
+				opts.RepoArg = args[0]
+			}
+			if runF != nil {
+				return runF(&opts)
+			}
+			return viewRun(&opts)
+		},
+	}
+
+	cmd.Flags().BoolVarP(&opts.Web, "web", "w", false, "Open a repository in the browser")
+
+	return cmd
+}
+
+func viewRun(opts *ViewOptions) error {
+	var toView ghrepo.Interface
+	if opts.RepoArg == "" {
+		var err error
+		toView, err = opts.BaseRepo()
```

Comment:
This part is tricky. The invocation here is correct, but I think that for backwards compatibility, we'll need to pass a different `BaseRepo` implementation to the current crop of commands that use the ResolveRemotesToRepos strategy. (Basically, all commands that are not `gh api` and `gh gist`.)

Right now, most of the commands that operate on the current repository use ResolveRemotesToRepos in order to figure out whether they should make requests against the current repository or its parent (if in a fork). We are considering moving away from this lookup strategy #924, but until we do, I think it's best to keep the current behavior.

Right now, the BaseRepo implementation that is passed through a Factory is the legacy base repo resolution backed by `fsContext.BaseRepo`, which only consults git remotes based on their names but performs no resolution via API requests. I've specifically chosen this strategy for `gh api` because I did not want any magic nor additional API requests in its implementation.

Reply:
I'll try and massage either determineBaseRepo or a copy of it into a form usable by isolated commands

Reply:
I'm not thrilled with what I did; mainly because the non-test factory version of `ResolvedBaseRepo` is not getting exercised in tests. What do you think?